src/test/java/com/rtb/projectmanagementtool/task/TaskControllerTest.java

```diff
@@ -66,111 +55,99 @@ public void tearDown() {
+    TaskData task2 = new TaskData(projectID2, name2, description2, status2, users2);
+    TaskData task3 = new TaskData(projectID3, name3, description3, status3, users3);
 
     // Assert no task entities are found
     Assert.assertEquals(0, ds.prepare(new Query("Task")).countEntities(withLimit(10)));
 
-    // Add task entities to ds
-    long taskID1 = ds.put(task1.toEntity()).getId();
-    long taskID2 = ds.put(task2.toEntity()).getId();
-    long taskID3 = ds.put(task3.toEntity()).getId();
+    // Add task entities with TaskController
+    taskController.addTasks(new ArrayList<>(Arrays.asList(task1, task2, task3)));
 
     // Assert 3 entities were added
     Assert.assertEquals(3, ds.prepare(new Query("Task")).countEntities(withLimit(10)));
-
-    // Get task with TaskController
-    TaskController taskController = new TaskController(ds);
-    TaskData getTask = taskController.getTaskByID(taskID2);
-
-    // Build expected task
-    TaskData task2 =
-        new TaskData(taskID2, projectID2, name2, description2, status2, users2, subtasks2);
-
-    // Assert task retrieved is correct
-    Assert.assertEquals("getTask", task2, getTask);
   }
 
   @Test
   public void testAddSubtasks() {
     DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
     TaskController taskController = new TaskController(ds);
 
-    // Add task1 and add task2 and task3 as its subtasks with TaskController
-    long taskID1 = taskController.addTasks(new ArrayList<>(Arrays.asList(task1))).get(0).getId();
-    ArrayList<Long> subtaskIDs =
-        taskController.getTaskIDsFromKeys(
-            taskController.addSubtasks(task1, new ArrayList<>(Arrays.asList(task2, task3))));
+    // Create tasks
+    TaskData task1 = new TaskData(projectID1, name1, description1, status1, users1);
+    TaskData task2 = new TaskData(projectID2, name2, description2, status2, users2);
+    TaskData task3 = new TaskData(projectID3, name3, description3, status3, users3);
+
+    // Add task1 with TaskController
+    taskController.addTasks(new ArrayList<>(Arrays.asList(task1)));
+
+    // Add task2 and task3 as subtasks of task1 with TaskController
+    ArrayList<TaskData> subtasks = new ArrayList<>(Arrays.asList(task2, task3));
+    taskController.addSubtasks(task1, subtasks);
 
     // Get subtasks
     Query query =
-        new Query("Task")
-            .addFilter("__key__", FilterOperator.IN, taskController.getKeysFromTaskIDs(subtaskIDs));
+        new Query("Task").addFilter("parentTaskID", FilterOperator.EQUAL, task1.getTaskID());
     PreparedQuery results = ds.prepare(query);
 
     // Assert correct subtasks were found
+    TaskData task;
     for (Entity entity : results.asIterable()) {
-      long taskID = entity.getKey().getId();
-      Assert.assertTrue("addSubtasks", subtaskIDs.contains(taskID));
+      task = new TaskData(entity);
+      Assert.assertTrue("addSubtasks", subtasks.contains(task));
```

Comment:
assertEquals can do list of objects I believe, so you can do something like `assertEquals(expected_subtasks, actual_subtasks)` instead of looping through them 1 by 1.  It sounds like more work (since you'll have to loop once to build actual_subtasks, then presumably assertEquals would loop as well), but in error scenarios it would print more details info like "expected contains tasks 1, 2, 3;  actual contains tasks 2, 3", compared to assertTrue which would only report back "result is false"